### PR TITLE
documented code

### LINKED ISSUE #7: code documentation

### PR DESCRIPTION
**Pull request by Invincible2211**
closes #7

### CODE REVIEW

File: src/main/java/de/yggdrasil/core/dal/data/datasources/ConfigDB.java [hunk 1]
```diff
@@ -3,16 +3,34 @@
 import de.yggdrasil.core.dal.data.DataSource;
 import de.yggdrasil.core.dal.data.datasources.repositorys.ConfigRepository;
 
+/**
+ * The {@code ConfigDB} class implements the {@code DataSource} interface with a type parameter of {@code String}.
+ * It provides methods to retrieve and write configuration data from a {@code ConfigRepository}.
+ *
+ * @param <String> the type of data stored in the data source
+ */
 public class ConfigDB implements DataSource<String> {
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/datasources/ConfigDB.java:12]: Maybe rename it to `ConfigDS` or `ConfigDataSource` as it's not really a data base?

File: src/main/java/de/yggdrasil/core/dal/data/datasources/RabbitMQ.java [hunk 2]
```diff
@@ -24,11 +28,21 @@ public class RabbitMQ implements EventDataSource<RabbitMQMessage, DALDefaultEven
     private Channel channel;
     private final DALEventbus eventbus = new DALDefaultEventBus();
 
+    /**
+     * The callback variable is a private final variable of type DeliverCallback.
+     * It is used to handle incoming messages from RabbitMQ.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/datasources/RabbitMQ.java:33]:
```suggestion
     * used to handle incoming messages from RabbitMQ.
```
The description of the field modifiers aswell as the description of the type is unnecessary.

File: src/main/java/de/yggdrasil/core/dal/data/datasources/RabbitMQ.java [hunk 3]
```diff
@@ -24,11 +28,21 @@ public class RabbitMQ implements EventDataSource<RabbitMQMessage, DALDefaultEven
...
+     */
     private final DeliverCallback callback = (consumerTag, delivery) -> {
         RabbitMQMessage message = RabbitMQPackageReader.readPackage(delivery.getBody());
         this.eventbus.triggerEvent(new RabbitMQDataReceivedEvent(this, message));
     };
 
+    /**
+     * The RabbitMQ constructor creates a RabbitMQ object with the specified queue name.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/datasources/RabbitMQ.java:41]:
```suggestion
     * Creates a RabbitMQ object with the specified queue name.
```

File: src/main/java/de/yggdrasil/core/dal/data/event/DALDefaultEventBus.java [hunk 4]
```diff
@@ -4,10 +4,20 @@
 import java.util.HashMap;
 import java.util.List;
 
+/**
+ * The DALDefaultEventBus class is an implementation of the DALEventbus interface.
+ * It provides methods for triggering events and registering event listeners.
+ */
 public class DALDefaultEventBus implements DALEventbus{
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/event/DALDefaultEventBus.java:11]: Either change it to `DALDefaultEventbus` or change the `DALEventbus` to match this, don't mix upper and lower case.

File: src/main/java/de/yggdrasil/core/dal/data/event/DALDefaultEventBus.java [hunk 5]
```diff
@@ -4,10 +4,20 @@
...
 
     private final HashMap<Class<? extends DataSourceDataEvent>, List<DataSourceDataListener>> eventListeners = new HashMap<>();
 
+    /**
+     * Triggers an event on the event bus, notifying all registered event listeners.
+     *
+     * @param event the data source data event to trigger
+     * @return true if the event was triggered successfully, false otherwise
+     */
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/event/DALDefaultEventBus.java:20]: You don't necessarily have to have these comments here as they are already present in the super type.

File: src/main/java/de/yggdrasil/core/dal/data/event/DALDefaultEventBus.java [hunk 6]
```diff
@@ -17,6 +27,11 @@ public boolean triggerEvent(DataSourceDataEvent event) {
         return true;
     }
 
+    /**
+     * Registers a listener for data events triggered by a data source.
+     *
+     * @param listener the listener to register
+     */
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/event/DALDefaultEventBus.java:34]: Same as above.

File: src/main/java/de/yggdrasil/core/dal/data/event/DALEventbus.java [hunk 7]
```diff
@@ -1,9 +1,24 @@
...
+ */
 public interface DALEventbus {
 
+    /**
+     * Triggers an event on the event bus, notifying all registered event listeners.
+     *
+     * @param event the data source data event to trigger
+     * @return true if the event was triggered successfully, false otherwise
+     */
     boolean triggerEvent(DataSourceDataEvent event);
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/event/DALEventbus.java:15]: Personally i think the term `fire` is better for events but it's up to you to decide

File: src/main/java/de/yggdrasil/core/dal/data/event/DataSourceDataListener.java [hunk 8]
```diff
@@ -2,9 +2,20 @@
...
     void handleEvent(T event);
 
+    /**
+     * Returns the class of the event associated with this listener.
+     *
+     * @param <T> the type of the event
+     * @return the class of the event
+     */
     default Class<T> getEventClass() {
         return (Class<T>) ((ParameterizedType) getClass().getGenericInterfaces()[0])
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/event/DataSourceDataListener.java:20]: Again might cause problems with raw types

File: src/main/java/de/yggdrasil/core/dal/data/event/events/RabbitMQDataReceivedEvent.java [hunk 9]
```diff
@@ -3,21 +3,43 @@
 import de.yggdrasil.core.dal.data.DataSource;
 import de.yggdrasil.core.dal.data.network.rabbitmq.RabbitMQMessage;
 
+/**
+ * The RabbitMQDataReceivedEvent class represents an event that is triggered when data is received from RabbitMQ.
+ * It implements the DataReceivedEvent interface.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/event/events/RabbitMQDataReceivedEvent.java:8]: Not required to say this, as it's obvious when looking at the class

File: src/main/java/de/yggdrasil/core/dal/data/event/events/RabbitMQDataReceivedEvent.java [hunk 10]
```diff
@@ -3,21 +3,43 @@
...
 public class RabbitMQDataReceivedEvent implements DataReceivedEvent{
 
     private final DataSource dataSource;
     private final RabbitMQMessage message;
 
+    /**
+     * The RabbitMQDataReceivedEvent class represents an event that is triggered when data is received from RabbitMQ.
+     * It implements the DataReceivedEvent interface.
+     *
+     * This event contains the data source from which the data is received and the RabbitMQ message that contains the received data.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/event/events/RabbitMQDataReceivedEvent.java:21]: This is just a repetition of the documentation that is already present above the class

File: src/main/java/de/yggdrasil/core/dal/data/event/events/RabbitMQDataReceivedEvent.java [hunk 11]
```diff
@@ -3,21 +3,43 @@
...
+     */
     public RabbitMQDataReceivedEvent(DataSource dataSource, RabbitMQMessage message){
         this.dataSource = dataSource;
         this.message = message;
     }
 
+    /**
+     * Retrieves the data source of the RabbitMQDataReceivedEvent.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/event/events/RabbitMQDataReceivedEvent.java:29]:
```suggestion
     * Retrieves the data source of the Event.
```

File: src/main/java/de/yggdrasil/core/dal/data/event/events/RabbitMQDataReceivedEvent.java [hunk 12]
```diff
@@ -3,21 +3,43 @@
...
+     *
+     * @return the data source from which the data is received
+     */
     @Override
     public DataSource getSource() {
         return this.dataSource;
     }
 
+    /**
+     * Retrieves the RabbitMQ message associated with the event.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/event/events/RabbitMQDataReceivedEvent.java:39]:
```suggestion
     * Retrieves the RabbitMQ message associated with the Event.
```

File: src/main/java/de/yggdrasil/core/dal/data/event/listeners/DataReceivedListener.java [hunk 13]
```diff
@@ -3,6 +3,13 @@
 import de.yggdrasil.core.dal.data.event.DataSourceDataEvent;
 import de.yggdrasil.core.dal.data.event.DataSourceDataListener;
 
+/**
+ * The DataReceivedListener interface represents a listener for data received events triggered by a data source.
+ * It extends the DataSourceDataListener interface and is a generic interface that takes a type parameter T.
+ * T must be a subclass of DataSourceDataEvent.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/event/listeners/DataReceivedListener.java:9]: Not required to say as again it's obvious when looking at the interface

File: src/main/java/de/yggdrasil/core/dal/data/network/rabbitmq/RabbitMQMessage.java [hunk 14]
```diff
@@ -1,3 +1,7 @@
 package de.yggdrasil.core.dal.data.network.rabbitmq;
 
+/**
+ * The RabbitMQMessage class represents a message in RabbitMQ.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/network/rabbitmq/RabbitMQMessage.java:4]: Im not sure if the `in` fits in this sentence

File: src/main/java/de/yggdrasil/core/dal/data/network/rabbitmq/RabbitMQPackage.java [hunk 15]
```diff
@@ -5,16 +5,29 @@
 
 import java.nio.ByteBuffer;
 
+/**
+ * Represents a RabbitMQ package that implements the Package interface.
+ * A RabbitMQPackage consists of a header and data.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/network/rabbitmq/RabbitMQPackage.java:10]:
```suggestion
 * Implementation of the Package interface for RabbitMQ.
```

File: src/main/java/de/yggdrasil/core/dal/data/network/rabbitmq/RabbitMQPackage.java [hunk 16]
```diff
@@ -5,16 +5,29 @@
...
+ */
 public class RabbitMQPackage implements Package {
 
     private final byte[] header;
     private final byte[] data;
 
+    /**
+     * Represents a RabbitMQ package that implements the Package interface.
+     * A RabbitMQPackage consists of a header and data.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/network/rabbitmq/RabbitMQPackage.java:19]: Just a repetition of the class level documentation

File: src/main/java/de/yggdrasil/core/dal/data/network/rabbitmq/RabbitMQPackage.java [hunk 17]
```diff
@@ -5,16 +5,29 @@
...
+     */
     public RabbitMQPackage(RabbitMQMessage message){
         header = ByteBuffer.allocate(4).putInt(message.key().length()).array();
         data = ByteUtils.connectByteArrays(message.key().getBytes(), message.content());
     }
 
+    /**
+     * Converts the RabbitMQPackage object to a byte array.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/data/network/rabbitmq/RabbitMQPackage.java:27]:
```suggestion
     * Converts the Package object to a byte array.
```

File: src/main/java/de/yggdrasil/core/dal/exceptions/DuplicateAdapterForClassException.java [hunk 18]
```diff
@@ -1,4 +1,7 @@
 package de.yggdrasil.core.dal.exceptions;
 
+/**
+ * Exception thrown when attempting to add a duplicate adapter for a class in the AdapterLibrary.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/exceptions/DuplicateAdapterForClassException.java:4]:
```suggestion
 * Exception for attempting to add a duplicate adapter for a class in the AdapterLibrary.
```

File: src/main/java/de/yggdrasil/core/dal/exceptions/MissingPipelineException.java [hunk 19]
```diff
@@ -2,6 +2,9 @@
 
 import de.yggdrasil.core.dal.requests.DALRequest;
 
+/**
+ * This class represents an exception that is thrown when there is no pipeline available for a request.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/exceptions/MissingPipelineException.java:6]:
```suggestion
 * Exception for when there is no pipeline available for a request.
```

File: src/main/java/de/yggdrasil/core/dal/pipeline/DALPipeline.java [hunk 20]
```diff
@@ -2,6 +2,44 @@
 
 import java.lang.annotation.*;
 
+/**
+ * The DALPipeline annotation is used to mark classes that represent a pipeline for reading
+ * or writing data from/to a data source.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/pipeline/DALPipeline.java:7]:
```suggestion
 * or writing data from or to a data source.
```

File: src/main/java/de/yggdrasil/core/dal/pipeline/DALPipeline.java [hunk 21]
```diff
@@ -2,6 +2,44 @@
...
+ *
+ * The annotation is meant to be used in combination with the Pipeline interface.
+ *
+ * The PipelineCollector interface can be used to collect all classes annotated with DALPipeline
+ * that are assignable from Pipeline.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/pipeline/DALPipeline.java:12]: This is wrong the `ClassCollector` does implement this behavior, not the interface.

File: src/main/java/de/yggdrasil/core/dal/pipeline/DALPipeline.java [hunk 22]
```diff
@@ -2,6 +2,44 @@
...
+ *
+ * Example Usage:
+ *
+ * // Interface definition
+ * @DALPipeline
+ * public interface Pipeline { }
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/pipeline/DALPipeline.java:18]: This isn't really an example usage

File: src/main/java/de/yggdrasil/core/dal/pipeline/DALPipelineProzessor.java [hunk 23]
```diff
@@ -14,6 +14,10 @@
 import java.lang.reflect.InvocationTargetException;
 import java.util.HashMap;
 
+/**
+ * The DALPipelineProzessor class represents a processor for data access layer (DAL) pipelines.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/pipeline/DALPipelineProzessor.java:18]:
```suggestion
 * The DALPipelineProzessor class represents a processor for DAL pipelines.
```
At this point you shouldn't have to explain what the abbreviation means.

File: src/main/java/de/yggdrasil/core/dal/pipeline/DALPipelineProzessor.java [hunk 24]
```diff
@@ -25,10 +29,24 @@ public class DALPipelineProzessor {
         setup();
     }
 
+    /**
+     * Initializes the pipeline collection by adding a ClassCollector instance.
+     * This method is used internally and is called during object initialization.
+     * It adds the ClassCollector instance to the pipeline collection.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/pipeline/DALPipelineProzessor.java:35]: You are repeating yourself here

File: src/main/java/de/yggdrasil/core/dal/pipeline/Pipeline.java [hunk 25]
```diff
@@ -5,5 +5,8 @@
 import de.yggdrasil.core.dal.requests.DALRequest;
 import de.yggdrasil.core.dal.requests.DALWriteRequest;
 
+/**
+ * The Pipeline interface represents a pipeline for reading or writing data from/to a data source.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/pipeline/Pipeline.java:9]:
```suggestion
 * The Pipeline interface represents a pipeline for reading or writing data from or to a data source.
```

File: src/main/java/de/yggdrasil/core/dal/pipeline/PipelineCollector.java [hunk 26]
```diff
@@ -2,8 +2,16 @@
 
 import java.util.Set;
 
+/**
+ * The PipelineCollector interface is responsible for collecting classes that are annotated with the `DALPipeline` annotation and are assignable from the `Pipeline` class.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/pipeline/PipelineCollector.java:6]: This behavior isn't implemented by the interface

File: src/main/java/de/yggdrasil/core/dal/pipeline/PipelineCollector.java [hunk 27]
```diff
@@ -2,8 +2,16 @@
...
+ */
 public interface PipelineCollector {
 
+    /**
+     * Collects classes that are annotated with the {@code DALPipeline} annotation and are assignable from the {@code Pipeline} class.
+     *
+     * @return a set of classes that implement the {@code Pipeline} interface
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/pipeline/PipelineCollector.java:13]: Again this behavior is only present in the implementation, but not in the interface

File: src/main/java/de/yggdrasil/core/dal/pipeline/implementations/DefaultConfigPipeline.java [hunk 28]
```diff
@@ -7,16 +7,31 @@
 import de.yggdrasil.core.dal.requests.implementations.ConfigWriteRequest;
 import de.yggdrasil.core.dal.responses.implementations.ConfigResponse;
 
+/**
+ * The DefaultConfigPipeline class represents a default implementation of a pipeline for reading and writing configuration data.
+ * It implements the ReadPipeline interface for reading data and the WritePipeline interface for writing data.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/pipeline/implementations/DefaultConfigPipeline.java:12]: This seems rather unnecessary as it's obvious when looking at the class hierarchy

File: src/main/java/de/yggdrasil/core/dal/requests/DALReadRequest.java [hunk 29]
```diff
@@ -3,10 +3,22 @@
 import de.yggdrasil.core.dal.data.DataSource;
 import java.lang.reflect.ParameterizedType;
 
+/**
+ * This interface represents a read request to a data source.
+ * It extends the DALRequest interface.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/requests/DALReadRequest.java:8]: Again seems unnecessary to mention as it's obvious

File: src/main/java/de/yggdrasil/core/dal/requests/DALReadRequest.java [hunk 30]
```diff
@@ -3,10 +3,22 @@
...
     String getIdentifier();
 
+    /**
+     * Retrieves the class of the data source.
+     *
+     * @param <T> the type of the data source
+     * @return the class of the data source
+     */
     default Class<T> getDatasourceClass() {
         return (Class<T>) ((ParameterizedType) getClass().getGenericInterfaces()[0])
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/requests/DALReadRequest.java:23]: Again might cause problems with raw usage

File: src/main/java/de/yggdrasil/core/dal/requests/DALRequest.java [hunk 31]
```diff
@@ -1,3 +1,7 @@
 package de.yggdrasil.core.dal.requests;
 
+/**
+ * This interface represents a data access layer (DAL) request.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/requests/DALRequest.java:4]: Again you shouldn't have to explain the abbreviation of the name

File: src/main/java/de/yggdrasil/core/dal/requests/DALWriteRequest.java [hunk 32]
```diff
@@ -4,12 +4,24 @@
 
 import java.lang.reflect.ParameterizedType;
 
+/**
+ * This interface represents a write request to a data source.
+ * It extends the DALRequest interface.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/requests/DALWriteRequest.java:9]: Again seems unnecassary as it's obvious

File: src/main/java/de/yggdrasil/core/dal/requests/DALWriteRequest.java [hunk 33]
```diff
@@ -4,12 +4,24 @@
...
 
     V getData();
 
+    /**
+     * Gets the class of the data source.
+     *
+     * @return the class of the data source
+     */
     default Class<T> getDatasourceClass() {
         return (Class<T>) ((ParameterizedType) getClass().getGenericInterfaces()[0])
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/requests/DALWriteRequest.java:26]: Again might cause problems with raw usage

File: src/main/java/de/yggdrasil/core/dal/requests/implementations/ConfigReadRequest.java [hunk 34]
```diff
@@ -3,6 +3,11 @@
 import de.yggdrasil.core.dal.data.datasources.ConfigDB;
 import de.yggdrasil.core.dal.requests.DALReadRequest;
 
+/**
+ * The {@code ConfigReadRequest} class represents a read request to the {@code ConfigDB} data source.
+ * It implements the {@code DALReadRequest} interface with {@code ConfigDB} as the data source type.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/requests/implementations/ConfigReadRequest.java:8]: This is unnecassary to mention as again it's obvious

File: src/main/java/de/yggdrasil/core/dal/requests/implementations/ConfigWriteRequest.java [hunk 35]
```diff
@@ -3,6 +3,13 @@
 import de.yggdrasil.core.dal.data.datasources.ConfigDB;
 import de.yggdrasil.core.dal.requests.DALWriteRequest;
 
+/**
+ * The `ConfigWriteRequest` class represents a write request to the `ConfigDB` data source.
+ * It implements the `DALWriteRequest` interface with `ConfigDB` as the data source type and `String` as the data type being written.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/requests/implementations/ConfigWriteRequest.java:8]: This is unnecassary to mention as again it's obvious

File: src/main/java/de/yggdrasil/core/dal/responses/DALResponse.java [hunk 36]
```diff
@@ -1,5 +1,10 @@
 package de.yggdrasil.core.dal.responses;
 
+/**
+ * The DALResponse interface represents a response from the DAL (Data Access Layer) containing data of type T.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/responses/DALResponse.java:4]: You don't have to explain the abbreviation

File: src/main/java/de/yggdrasil/core/dal/responses/implementations/ConfigResponse.java [hunk 37]
```diff
@@ -2,4 +2,8 @@
 
 import de.yggdrasil.core.dal.responses.DALResponse;
 
+/**
+ * The ConfigResponse class represents a response from the DAL (Data Access Layer) that contains configuration data of type String.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/responses/implementations/ConfigResponse.java:6]: You don't have to explain the abbreviation

Also i don't think that the meaning of the sentence is clear here, maybe rephrase it

File: src/main/java/de/yggdrasil/core/dal/responses/implementations/ConfigResponse.java [hunk 38]
```diff
@@ -2,4 +2,8 @@
...
+ * It implements the DALResponse interface with String as the type parameter.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/responses/implementations/ConfigResponse.java:7]: Also this is unnecessary to mention as it's obvious

File: src/main/java/de/yggdrasil/core/dal/strings/DatabaseStrings.java [hunk 39]
```diff
@@ -1,5 +1,8 @@
 package de.yggdrasil.core.dal.strings;
 
+/**
+ * This interface provides string constants related to databases and their configurations.
+ */
 public interface DatabaseStrings {
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/strings/DatabaseStrings.java:6]: Maybe rename to `DatabaseConstants`?

File: src/main/java/de/yggdrasil/core/dal/strings/logging/AdapterLibraryLogger.java [hunk 40]
```diff
@@ -1,5 +1,8 @@
 package de.yggdrasil.core.dal.strings.logging;
 
+/**
+ * The AdapterLibraryLogger interface provides constants for logging messages related to AdapterLibrary.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/strings/logging/AdapterLibraryLogger.java:4]:
```suggestion
 * The AdapterLibraryLogger provides constants for logging messages related to the AdapterLibrary.
```

File: src/main/java/de/yggdrasil/core/dal/strings/logging/DALLogger.java [hunk 41]
```diff
@@ -1,5 +1,25 @@
 package de.yggdrasil.core.dal.strings.logging;
 
+/**
+ * DALLogger is an interface that provides constants for logging messages related to the DAL (Data Access Layer) operations.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/strings/logging/DALLogger.java:4]:
```suggestion
 * DALLogger provides constants for logging messages related to the DAL operations.
```
again no need to explain the abbreviation

File: src/main/java/de/yggdrasil/core/dal/strings/logging/DALLogger.java [hunk 42]
```diff
@@ -1,5 +1,25 @@
...
+ * It contains the following constants:
+ * - DAL_SAVE: A log message indicating a DAL save request, with a placeholder for the request class name.
+ * - DAL_READ: A log message indicating a DAL load request, with a placeholder for the request class name.
+ * - DAL_EXCEPTION_PIPELINE: A log message indicating that no pipeline is configured for a request, with a placeholder for the request class name.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/strings/logging/DALLogger.java:8]: Maybe explain these documentations at field level and not at class level?

File: src/main/java/de/yggdrasil/core/dal/strings/logging/DALPipelineProzessorLogger.java [hunk 43]
```diff
@@ -1,5 +1,8 @@
 package de.yggdrasil.core.dal.strings.logging;
 
+/**
+ * The DALPipelineProzessorLogger interface defines the log messages to be used by the DALPipelineProzessor class.
+ */
 public interface DALPipelineProzessorLogger {
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/strings/logging/DALPipelineProzessorLogger.java:6]:
```suggestion
public interface DALPipelineProzessorLoggerMessages {
```

File: src/main/java/de/yggdrasil/core/dal/strings/logging/DatasourceLibraryLogger.java [hunk 44]
```diff
@@ -1,5 +1,8 @@
 package de.yggdrasil.core.dal.strings.logging;
 
+/**
+ * The DatasourceLibraryLogger interface provides a constant string for logging the number of data sources added to the library.
+ */
 public interface DatasourceLibraryLogger {
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/strings/logging/DatasourceLibraryLogger.java:6]:
```suggestion
public interface DatasourceLibraryLoggerMessages {
```

File: src/main/java/de/yggdrasil/core/dal/utils/ByteUtils.java [hunk 45]
```diff
@@ -1,7 +1,17 @@
 package de.yggdrasil.core.dal.utils;
 
+/**
+ * This class provides utility methods for working with byte arrays.
+ */
 public class ByteUtils {
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/utils/ByteUtils.java:6]:
```suggestion
public class ByteArrayUtils {
```

File: src/main/java/de/yggdrasil/core/dal/utils/ByteUtils.java [hunk 46]
```diff
@@ -1,7 +1,17 @@
...
 
+    /**
+     * Concatenates two byte arrays into a single byte array.
+     *
+     * @param arrayA the first byte array
+     * @param arrayB the second byte array
+     * @return a new byte array containing the concatenated elements of arrayA followed by arrayB
+     */
     public static byte[] connectByteArrays(byte[] arrayA, byte[] arrayB){
         byte[] result = new byte[arrayA.length+arrayB.length];
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/utils/ByteUtils.java:16]: Maybe use the following code instead.
```java
    byte[] newArray = Arrays.copyOf(arrayA, arrayA.length + arrayB.length);
    System.arraycopy(arrayB, 0, newArray, arrayA.length, arrayB.length);
    return newArray;
```

File: src/main/java/de/yggdrasil/core/dal/utils/ClassCollector.java [hunk 47]
```diff
@@ -17,11 +17,23 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
+/**
+ * The ClassCollector class is responsible for collecting classes annotated with specific annotations.
+ * It implements default implementations for the AdapterCollector, PipelineCollector, and DatasourceCollector interfaces.
```
Comment (Morazzer) [inline src/main/java/de/yggdrasil/core/dal/utils/ClassCollector.java:22]: this is a rather weird sentence as you are saying the same thing twice.

**Review by Morazzer — CHANGES_REQUESTED** [review]
You might also want to extract common methods to a either interface with a default implementation or to a Utility class

**Comment by Morazzer** [comment]
Also you could have used a [ci skip tag](https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs) for this pr, but that is just a side note.

File: src/main/java/de/yggdrasil/core/dal/utils/ClassCollector.java (hunk 47 above)
Comment (Invincible2211) [reply src/main/java/de/yggdrasil/core/dal/utils/ClassCollector.java:22]: naja, es gibt ja die allgemeine Funktion ( collect(...) ) und die spezifischen Interface implementations, also ist imo nicht gedoppelt

**Review by Morazzer — APPROVED** [review]
